pkg/controller/localvolumediscovery/localvolumediscovery_controller.go

```diff
@@ -283,13 +328,15 @@ func getOwnerRefs(cr *localv1alpha1.LocalVolumeDiscovery) []metav1.OwnerReferenc
 	}
 }
 
-func getUIDEnvVar() corev1.EnvVar {
-	return corev1.EnvVar{
-		Name: "UID",
-		ValueFrom: &corev1.EnvVarSource{
-			FieldRef: &corev1.ObjectFieldSelector{
-				FieldPath: "metadata.uid",
-			},
+func getEnvVars(objName, uid string) []corev1.EnvVar {
+	return []corev1.EnvVar{
+		{
+			Name:  "UID",
```

Comment:
Can this environment variable be suitable named? `UID` as a generic name can conflict with other things.

Reply:
Updated name to `DISCOVERY_OBJECT_UID`